test/lib/ansible_test/_data/requirements/sanity.txt

```diff
@@ -4,6 +4,7 @@ pycodestyle ; python_version >= '3.5' # only used on python 3.5+
 pylint ; python_version >= '3.5' and python_version < '3.9' # only used on python 3.5+ and does not yet support python 3.9
 pyyaml
 rstcheck ; python_version >= '3.5' # only used on python 3.5+
+semantic_version ; python_version >= '3.5' # only used on python 3.5+
```

Comment:
I just realized we should be using the same `SemanticVersion` code that `ansible-galaxy` is using:

https://github.com/ansible/ansible/blob/98fbd1b72f501fbdee8388e95b02c2582717971a/lib/ansible/utils/version.py#L128

Reply:
Interesting, I didn't knew Ansible had it's own semantic version code. Anyway, while looking over it I found a bug I think. I'll adjust this PR to use the class and then I'll create a PR to fix the bug.

Reply:
Bugfix in #69395.